Composer/packages/client/src/store/action/lg.ts

```diff
@@ -1,52 +1,45 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 import clonedeep from 'lodash/cloneDeep';
-import debounce from 'lodash/debounce';
 
 import { ActionTypes } from '../../constants';
-import httpClient from '../../utils/httpUtil';
 import * as lgUtil from '../../utils/lgUtil';
 import { undoable } from '../middlewares/undo';
-import { ActionCreator, State } from '../types';
+import { ActionCreator, State, Store } from '../types';
 
-import { setError } from './error';
+export const updateLgFile: ActionCreator = async (store, { id, content }) => {
+  store.dispatch({
+    type: ActionTypes.UPDATE_LG,
+    payload: { id, content },
+  });
+};
 
-//remove editor's debounce and add it to action
-export const debouncedUpdateLg = debounce(async (store, id, projectId, content, lastModified) => {
-  try {
-    const response = await httpClient.put(`/projects/${projectId}/lgFiles/${id}`, {
-      id,
-      projectId,
-      content,
-      lastModified,
-    });
-    store.dispatch({
-      type: ActionTypes.UPDATE_TIMESTAMP,
-      payload: {
-        id: id,
-        type: 'lg',
-        lastModified: response.data.lastModified,
-      },
-    });
-  } catch (err) {
-    setError(store, {
-      status: err.response.status,
-      message: err.response && err.response.data.message ? err.response.data.message : err,
-      summary: 'UPDATE LG ERROR',
-    });
+export const createLgFile: ActionCreator = async (store, { id, content }) => {
+  const lgFiles = store.getState().lgFiles;
+  const lgFile = lgFiles.find(lg => lg.id === id);
+  if (lgFile) {
+    throw new Error(`${id} lg file already exist`);
   }
-}, 500);
-
-export const updateLgFile: ActionCreator = async (store, { id, projectId, content }) => {
-  const state = store.getState();
-  const file = state.lgFiles.find(l => l.id === id);
-  if (file) {
-    store.dispatch({
-      type: ActionTypes.UPDATE_LG_SUCCESS,
-      payload: { id, projectId, content, lastModified: file.lastModified },
-    });
-    debouncedUpdateLg(store, id, projectId, content, file.lastModified);
+  // slot with common.lg import
+  let lgInitialContent = '';
+  const lgCommonFile = lgFiles.find(({ id }) => id === 'common');
```

Comment:
lgFile's id is included locale, e.g
lgFiles:
[
'common.en-us',
'common.zh-ch',
...
]